presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/Util.java

```diff
@@ -62,4 +67,23 @@ public static PlanNode transpose(PlanNode parent, PlanNode child)
                 parent.replaceChildren(
                         child.getSources())));
     }
+
+    /**
+     * @return If the node has outputs not in permittedOutputs, returns an identity projection containing only those node outputs also in permittedOutputs.
+     */
+    public static Optional<PlanNode> restrictOutputs(PlanNodeIdAllocator idAllocator, PlanNode node, Set<Symbol> permittedOutputs)
```

Comment:
The suggestion didn't pan out well.  In uses of this function, the return value is often combined with other plan nodes, for example using Optional.orElse().  Such uses work with Optional<PlanNode> but not Optional<ProjectNode>.  In contrast, the underlying Project-ness is not interesting.